src/openms/source/ANALYSIS/OPENSWATH/MRMIonSeries.cpp

```diff
@@ -421,9 +435,10 @@ namespace OpenMS
         }
       }
 
-      if (tr.getProduct().getChargeState() >= 1 && std::find(fragment_charges.begin(), fragment_charges.end(), tr.getProduct().getChargeState()) != fragment_charges.end())
+      if (fragment_charge >= 1 && 
```

Comment:
why use `tr.getProduct().getChargeState()` before and now use `fragment_charges`, I am a bit confused about why we have two charge states here?